test/clj/puppetlabs/trapperkeeper/services/webserver/jetty9_service_test.clj

```diff
@@ -729,7 +712,34 @@
           (let [response (http-client-common/get async-client "http://localhost:8080/hello" {:as :text})]
             @in-request-handler
             (tk-app/stop app)
-            (is (not (nil? (:error @response))))))))))
+            (is (not (nil? (:error @response)))))))))
+
+  (testing "tk app can still restart even if stop timeout expires"
+    (let [in-request-handler (promise)
+          sleepy-service (tk-core/service
+                          [[:WebserverService add-ring-handler]]
+                          (init [this context]
+                                (add-ring-handler
+                                 (fn [_]
+                                   (deliver in-request-handler true)
+                                   (Thread/sleep 2000)
```

Comment:
We only need the delay to happen for the first `get` request right?  For the second one, I was thinking we wouldn't want a delay anyway - just slows things down.  If that's true, seems like just delivering on the initial promise would be sufficient and the deref done in the handler the second request would effectively be a no-op because it was delivered for the first one already.

Is it a problem for this test if we're waiting in the ring handler function for the blocking `restart` to finish?  The `restart` should still finish, right?  And not throw an exception anymore per the previous code changes we merged?  Might leave the state of the first request in a somewhat ambiguous state, I suppose.  Not sure if it will have been interrupted on waiting for the promise by Jetty sending a thread interrupt during the shutdown?  Either way, seems like the original request will still fail from the client perspective by virtue of the server being torn down during Jetty's stop.